polars/polars-lazy/polars-plan/src/dsl/function_expr/list.rs

```diff
@@ -141,13 +141,23 @@ pub(super) fn concat(s: &mut [Series]) -> PolarsResult<Option<Series>> {
     let mut first = std::mem::take(&mut s[0]);
     let other = &s[1..];
 
-    let first_ca = match first.list().ok() {
+    let mut first_ca = match first.list().ok() {
         Some(ca) => ca,
         None => {
             first = first.reshape(&[-1, 1]).unwrap();
             first.list().unwrap()
         }
     };
+
+    let new_ca;
+    if first_ca.len() == 1 && !other.is_empty() {
+        let max_len = other.iter().map(|s| s.len()).max().unwrap();
+        if max_len > 1 {
+            new_ca = first_ca.new_from_index(0, max_len);
+            first_ca = &new_ca;
```

Comment:
Is there a nicer Rust pattern for handling this reassignment? The compiler really wanted this new series to have a scope outside the if statement to let me reassign it.

Reply:
That's because `new_ca` would get dropped when the scope ends.

You can clone on line 144. Then you can assign directly

Reply:
Ah! Thanks for that explanation. Still new to Rust. Does `to_owned()` offer any benefit over `clone()` there? I'm not sure how expensive a `clone()` on a `ChunkedArray`. Does it end up doing a deep copy of all it's items or just the wrapping type?

Reply:
> Does it end up doing a deep copy of all it's items or just the wrapping type?

The data is wrapped by an `Arc` so it is a cheap clone.